src/lib/core/Texture.ts

```diff
@@ -333,12 +342,15 @@ class Texture {
     }
 
     if (this.image || this.data) {
-      if (this.image && this.image.hasOwnProperty('width')) {
+      if (this.image && 'width' in this.image) {
```

Comment:
Sometimes images can have a width of 0, which is still valid. This will cause issues with that case, I think.

Reply:
This is exactly the same check it asserts the existence of the property and not the value but in this "shape" typescript can narrow the type on the subsequent checks.